tests/test_sftpserver.cpp

```diff
@@ -802,6 +804,68 @@ TEST_F(SftpServer, handles_reads)
     ASSERT_THAT(num_calls, Eq(1));
 }
 
+TEST_F(SftpServer, handle_extended_link)
+{
+    mpt::TempDir temp_dir;
+    auto file_name = temp_dir.path() + "/test-file";
+    auto link_name = temp_dir.path() + "/test-link";
+    make_file_with_content(file_name);
+
+    auto sftp = make_sftpserver();
+    auto msg = make_msg(SFTP_EXTENDED);
+    std::string extended_method("hardlink@openssh.com");
+    auto submessage = name_as_char_array(extended_method);
```

Comment:
This could just be auto submessage = name_as_char_array("hardlink@openssh.com");